.github/workflows/java.yml

```diff
@@ -0,0 +1,134 @@
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      SNAPSHOT:
+        type: boolean
+        description: Set SNAPSHOT true to publish
+
+jobs:
+  lint_markdown_files:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '2.6'
+        bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+    - name: Install gem
+      run: |
+        gem install awesome_bot
+    - name: Run tests
+      run: find . -type f -name '*.md' -exec awesome_bot {} \;
+  
+  integration_tests:
+    uses: optimizely/java-sdk/.github/workflows/integration_test.yml@mnoman/gitAction
+    with:
+      BUILD_NUMBER: ${{ github.run_id }}
+      GITHUB_CONTEXT: ${{ toJson(github) }}
+      EVENT_NAME: ${{ github.event_name }}
+      REPOSITORY: ${{ github.repository }}
+      SHA: ${{ github.sha }}
+      TRAVIS_PULL_REQUEST_SHA: ${{ github.event.pull_request.head.sha }}
+      TRAVIS_PULL_REQUEST: ${{ github.event.pull_request.number }}
+      EVENT_MESSAGE: ${{ github.event.message }}
+      HEAD_REF: ${{ github.head_ref }}
+      REF_NAME: ${{ github.ref_name }}
+    secrets:
+      CI_USER_TOKEN: ${{ secrets.CI_USER_TOKEN }}
+      TRAVIS_COM_TOKEN: ${{ secrets.TRAVIS_COM_TOKEN }}
```

Comment:
I see factoring out to "integration_test.yml" looks good, but not much gain (more overhead than sharing). I'll leave it up to you. I guess you already tried, but wondering if we cannot use ${{ github }} directly in "integration_test.yml" instead of passing as input parameters.

Reply:
Yeah, i actually asked for it that cleans-up our main.yaml file. 
Regarding your suggestion we tried but github is very specific to passing secrets info to actions, that's why this was not done.